functions/venue.js

```diff
@@ -476,23 +476,36 @@ exports.createVenue_v2 = functions.https.onCall(async (data, context) => {
 
 exports.upsertRoom = functions.https.onCall(async (data, context) => {
   checkAuth(context);
+
+  const result = {
+    provided: data,
+  };
+
   const { venueId, roomIndex, room } = data;
   await checkUserIsOwner(venueId, context.auth.token.user_id);
   const doc = await admin.firestore().collection("venues").doc(venueId).get();
 
   if (!doc || !doc.exists) {
     throw new HttpsError("not-found", `Venue ${venueId} not found`);
   }
-  const docData = doc.data();
-  let rooms = docData.rooms;
 
-  if (typeof roomIndex !== "number") {
-    rooms = [...rooms, room];
+  result.rooms = doc.data().rooms ?? [];
+
+  if (typeof roomIndex === "number") {
+    result.rooms[roomIndex] = room;
+    result.roomIndex = roomIndex;
   } else {
-    rooms[roomIndex] = room;
+    result.rooms = [...result.rooms, room];
+    result.roomIndex = result.rooms.length - 1;
   }
 
-  admin.firestore().collection("venues").doc(venueId).update({ rooms });
+  result.writeResult = await admin
```

Comment:
What functions are being returned?

Reply:
I miss-spoke, the result from the function is not used anywhere and it can be removed.

Reply:
If you really need this data you should use type assertion before returning it. Ex: you can see `get-badges` and the way they add type to the snapshots. `visitDoc.data() as UserVisit`

Reply:
TL;DR: `writeResult` only has one timestamp field `writeTime` which doesn't provide much use, so I will remove it.


Attempt to clear up possible confusion:

The file name where `upsertRoom` is located is `./functions/venue.js` and by it I'm guessing it's being run server side on a Node 10 without TypeScript support.

The comparative's name is `./scripts/get-badges.ts` and by it I'm guessing a client (non-web) CLI script transpiled from TypeScript before executed.

I think it would be OOS to this PR try to put TypeScript assertions into a server side JavaScript file.


On the "type assertion" note, one is already done,
as "safe" as it can `as CreateRoomResult ` be,
by declaring the return type of 

```typescript
export const createRoom = async (
  input: RoomInput_v2,
  venueId: string,
  user: UserInfo
): Promise<CreateRoomResult> => {
```
in `./src/api/admin.ts` with the interface defined as: 
```typescript
export interface CreateRoomResult {
  data: {
    provided: {
      room: Partial<Room>;
      roomIndex: number | unknown;
      venueId: string;
    };
    roomIndex: number;
    rooms: Room[];
    writeResult: admin.firestore.WriteResult;
  };
}
```

Reply:
removed `writeResult` in 1bb7aaa7c18844dca1014d68f039d8237c4bbf4d